### PR TITLE
AVRO-3330: Avrogen returns success if help requested

### PR DESCRIPTION
**Pull request by zcsizmadia**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO-3330) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AVRO-3330
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [X] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Review by KyleSchoonover — CHANGES_REQUESTED** [review]
Please review comments

File: lang/csharp/src/apache/codegen/AvroGen.cs [hunk 1]
```diff
@@ -29,7 +29,8 @@ static int Main(string[] args)
             if (args.Length == 0 || args[0] == "-h" || args[0] == "--help")
             {
                 Usage();
-                return 1;
+                // Return success if help requested
+                return args.Length == 0 ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (zcsizmadia) [reply lang/csharp/src/apache/codegen/AvroGen.cs]: I am not sure I agree with that statement: double the code for  a very simple  condition. However I modded the PR

**Review by KyleSchoonover — APPROVED** [review]
Looks good to me.  Thanks for making the updates.

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you for the contribution, @zcsizmadia !

**Pull request merged**